service/interpreter/persistence.go

```diff
@@ -158,16 +159,22 @@ func (am *PersistenceManager) LoadDataObjects(
 
 func (am *PersistenceManager) GetAllSearchAttributes() []iwfidl.SearchAttribute {
 	var res []iwfidl.SearchAttribute
-	for _, value := range am.searchAttributes {
-		res = append(res, value)
+	// NOTE: using DeterministicKeys so that the JSON snapshot for continueAsNew is stable for pagination
+	// TODO: we should use DeterministicKeys for every map iteration in interpreter for safety
+	// https://github.com/indeedeng/iwf/issues/510
+	for _, k := range DeterministicKeys(am.searchAttributes) {
```

Comment:
This is necessary otherwise the snapshot will have different order for different pages. 
It wasn't a problem becase we didn't see more than one page for continueAsNew.